From 890a9867b54dd3d1543c79c150fa9fcbbe997adf Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Tue, 3 Jun 2014 23:39:47 +1000
Subject: [PATCH 013/945] =?UTF-8?q?Don=E2=80=99t=20add=20port=20number=20to=20?=
 =?UTF-8?q?myth=20URL=20if=20it=E2=80=99s=20the=20default=20port,=20makes=20?=
 =?UTF-8?q?for=20cleaner=20URLs?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Also put IPv6 URL in lower-case, I find it prettier
---
 mythtv/libs/libmythbase/mythcorecontext.cpp |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mythtv/libs/libmythbase/mythcorecontext.cpp b/mythtv/libs/libmythbase/mythcorecontext.cpp
index 27cf6e5..ebd0b54 100644
--- a/mythtv/libs/libmythbase/mythcorecontext.cpp
+++ b/mythtv/libs/libmythbase/mythcorecontext.cpp
@@ -685,11 +685,11 @@ QString MythCoreContext::GenMythURL(QString host, int port, QString path, QStrin
 
 #if !defined(QT_NO_IPV6)
     // Basically if it appears to be an IPv6 IP surround the IP with [] otherwise don't bother
-    if (( addr.protocol() == QAbstractSocket::IPv6Protocol ) || (host.contains(":")))
-        m_host = "[" + host + "]";
+    if (addr.protocol() == QAbstractSocket::IPv6Protocol)
+        m_host = "[" + addr.toString().toLower() + "]";
 #endif
 
-    if (port > 0)
+    if (port > 0 && port != 6543)
         m_port = QString(":%1").arg(port);
     else
         m_port = "";
-- 
1.7.10.2

